### ISSUE TITLE
Posting Large file Gateway Timeout

### ISSUE THREAD

**Issue by bibinpic** [apiman/apiman#847]
Posting large file throws below error on Gateway and REST Client received 504 . 

{
    "responseCode": 504,
    "message": "The timeout period of 120000ms has been exceeded while executing POST /dctm-rest/repositories/ecmsrmr/folders/0b9bfda380282324/documents.json for host abc.local",
    "trace": "io.apiman.gateway.engine.beans.exceptions.ConnectorException: The timeout period of 120000ms has been exceeded while executing POST /dctm-rest/repositories/ecmsrmr/folders/0b9bfda380282324/documents.json for host vna-v-ny.glbxnet.local\n\tat io.apiman.gateway.platforms.vertx3.connector.HttpConnector$ExceptionHandler.handle(HttpConnector.java:319)\n\tat io.apiman.gateway.platforms.vertx3.connector.HttpConnector$ExceptionHandler.handle(HttpConnector.java:316)\n\tat io.vertx.core.http.impl.HttpClientRequestBase.handleException(HttpClientRequestBase.java:139)\n\tat io.vertx.core.http.impl.HttpClientRequestBase.timeout(HttpClientRequestBase.java:211)\n\tat io.vertx.core.http.impl.HttpClientRequestBase.handleTimeout(HttpClientRequestBase.java:194)\n\tat io.vertx.core.http.impl.HttpClientRequestBase.lambda$setTimeout$0(HttpClientRequestBase.java:124)\n\tat io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:907)\n\tat io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:866)\n\tat io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)\n\tat io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)\n\tat io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:462)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: io.vertx.core.http.impl.HttpClientRequestBase$1: The timeout period of 120000ms has been exceeded while executing POST /dctm-rest/repositories/ecmsrmr/folders/0b9bfda380282324/documents.json for abc.locall\n"
}

**Comment by bekihm** [comment]
Hello @bibinpic 

Maybe you can set [in the ConnectorFactory](https://github.com/apiman/apiman/blob/8e8fc6d8a64b270ed536b1df0ffd43137b89f854/gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/connector/ConnectorFactory.java#L91)
`.setConnectionTimeout(0)` 

And then try out if this solves the problem.

**Comment by tevosouza** [comment]
Hello!

@bekihm I'm thinking about to implement another "tab" into api implementation like an "Advanced Configurations" that permits some specific things for one specific api.
Like 'timeouts', 'strip or not keys (move it from endpoint tab)' maybe other things like edit query parameters

@bekihm , @volkflo , @msavy , @EricWittmann any thoughts?

**Comment by bekihm** [comment]
@tevosouza sounds very nice :-)

**Comment by EricWittmann** [comment]
I agree - that sounds nice.  The ability to override a global connection timeout for a specific API would be ideal.  Global defaults should probably be sane. :)

**Comment by tevosouza** [comment]
Hi @bibinpic,
After a few discussions, I realized that the 'advanced tab' is not interesting at the moment. To solve your problem @wbeck-fr implemented a custom policy that configures timeouts, take a look into [apiman/apiman-plugins#100](https://github.com/apiman/apiman-plugins/pull/100)
